tests/app/UnitTests/GitUI.Tests/Editor/Diff/AnsiEscapeUtilitiesTest.ParseEscape.cs

```diff
@@ -151,4 +151,71 @@ black to the end
+            {_escape_sequence}0;1;31m merge adjacent segments{_escape_sequence}m{_escape_sequence}0;31;1m continue{_escape_sequence}m{_escape_sequence}1;31m here{_escape_sequence}m
+            {_escape_sequence}0;2;31m ignore empty{_escape_sequence}m{_escape_sequence}0;2;31m{_escape_sequence}m
+            {_escape_sequence}0;1;31m ignore carriage return color {_escape_sequence}m{_escape_sequence}0;1;31m{carriageReplacer}{_escape_sequence}m
+            {_escape_sequence}0;1;31m but not space{_escape_sequence}m{_escape_sequence}0;31;1m {_escape_sequence}m
+            {_escape_sequence}0;31m merge ending with carriage return newline{_escape_sequence}m{carriageReplacer}
+            {_escape_sequence}0;31m some text{_escape_sequence}m
+            """.ReplaceLineEndings("\n").Replace(carriageReplacer, carriageReturn);
+        string expected_text = $"""
+            Text with red over
+            several lines
+             single newline with sequence
+             merge adjacent segments continue here
+             ignore empty
+             ignore carriage return color {carriageReplacer}
+             but not space 
+             merge ending with carriage return newline{carriageReplacer}
+             some text
+            """.ReplaceLineEndings("\n").Replace(carriageReplacer, carriageReturn);
+        StringBuilder sb = new();
+        List<TextMarker> textMarkers = [];
+
+        AnsiEscapeUtilities.ParseEscape(in_text, sb, textMarkers);
+
+        sb.ToString().Should().Be(expected_text);
+        textMarkers.Should().HaveCount(6);
+
+        textMarkers[0].Offset.Should().Be(10);
+        textMarkers[0].Length.Should().Be(23);
+        textMarkers[0].Color.Should().Be(SystemColors.Window);
+        textMarkers[0].ForeColor.Should().Be(Color.FromArgb(255, 255, 0, 0));
+
+        textMarkers[1].Offset.Should().Be(62);
+        textMarkers[1].Length.Should().Be(1);
+        textMarkers[1].Color.Should().Be(SystemColors.Window);
+        textMarkers[1].ForeColor.Should().Be(_redAnsiTheme[0]);
+
+        textMarkers[2].Offset.Should().Be(63);
+        textMarkers[2].Length.Should().Be(38);
+        textMarkers[2].Color.Should().Be(SystemColors.Window);
+        textMarkers[2].ForeColor.Should().Be(_redAnsiTheme[2]);
+
+        textMarkers[3].Offset.Should().Be(102);
+        textMarkers[3].Length.Should().Be(14);
+        textMarkers[3].Color.Should().Be(SystemColors.Window);
+        textMarkers[3].ForeColor.Should().Be(_redAnsiTheme[1]);
+
+        textMarkers[4].Offset.Should().Be(116);
+        textMarkers[4].Length.Should().Be(47);
+        textMarkers[4].Color.Should().Be(SystemColors.Window);
+        textMarkers[4].ForeColor.Should().Be(_redAnsiTheme[2]);
+
+        textMarkers[5].Offset.Should().Be(164);
+        textMarkers[5].Length.Should().Be(54);
+        textMarkers[5].Color.Should().Be(SystemColors.Window);
+        textMarkers[5].ForeColor.Should().Be(_redAnsiTheme[0]);
```

Comment:
Could all of these be verified with a single snapshot? That is, use Verify?

Reply:
The use of themes makes this difficult. Values need to be hardcoded in the theme.
(I have considered this as it tedious to make changes to the tests...)
Maybe later.